libvast/vast/json.hpp

```diff
@@ -234,6 +235,44 @@ json to_json(const T& x, Opts&&... opts) {
 
 json::object combine(const json::object& lhs, const json::object& rhs);
 
+namespace detail {
+
+template <class F>
+void each_field_impl(const json& x, F f, std::string& prefix) {
```

Comment:
Instead of using  a `std::string` here, you could consider `detail::stack_vector` with a `std::string`, similar to `record_type::each`.